### PR TITLE
Fix enrichment of nested Pydantic BaseModel

### PR DESCRIPTION
**Pull request by disrupted**

### CODE REVIEW

File: kpops/component_handlers/kubernetes/model.py [hunk 1]
```diff
@@ -12,7 +12,10 @@
 
 
 # JSON values
-Json: TypeAlias = dict[str, "Json"] | list["Json"] | str | int | float | bool | None
+JsonPrimitive: TypeAlias = (
+    dict[str, "Json"] | list["Json"] | str | int | float | bool | None
+)
+Json = TypeVar("Json", bound=JsonPrimitive)
```
Comment (disrupted) [inline kpops/component_handlers/kubernetes/model.py:18]: todo: move somewhere more generic

Comment (sujuka99) [reply kpops/component_handlers/kubernetes/model.py:18]: I could see it living in `kpops.utils`

Comment (disrupted) [reply kpops/component_handlers/kubernetes/model.py]: yep, I improved this quite a bit and moved it to `kpops.utils.types`

**Review by sujuka99 — APPROVED** [review]
LGTM!

**Pull request merged**